File: go/grind75/majority_element.go
package grind75

func majorityElement(nums []int) int {
	vote, majority := 0, 0

	for _, num := range nums {
		if vote == 0 {
			majority = num
		}
		if majority == num {
			vote++
		} else {
			vote--
		}
	}
	return majority
}

File: src/main/java/grind75/MajorityElement.java
package grind75;

public class MajorityElement {
    
    // ex: [3, 2, 3] -> 3
    public int majorityElement(int[] nums) {
        int vote =0, majority = 0;
        for (int num : nums) {
            if (vote == 0) {
                majority = num;
            }
            if (majority == num) {
               vote++;
            } else {
                vote--;
            }
        }
        return majority;   
    }
}
